### PR TITLE
fix #2392

### LINKED ISSUE #2392: find_MAP bug with start value and Uniform distribution
It looks like find_MAP has a problem with a start value when a variable has a `Uniform` distribution, as shown in the example below (version 3.1). It seems related to #2042.

```python
import numpy as np
import pymc3 as pm
from theano import as_op
import theano.tensor as tt

# Initialize random number generator
np.random.seed(123)

# True parameter values
alpha, sigma = 1, 1
beta = [1, 2.5]

# Size of dataset
size = 100

# Predictor variable
X1 = np.random.randn(size)
X2 = np.random.randn(size) * 0.2

# Simulate outcome variable
Y = alpha + beta[0]*X1 + beta[1]*X2 + np.random.randn(size)*sigma

basic_model = pm.Model()
with basic_model:

    # Priors for unknown model parameters

    # this works fine:
    #alpha = pm.Normal('alpha', mu=0, sd=10)
    #beta0 = pm.Normal('beta0', mu=0, sd=10)
    #beta1 = pm.Normal('beta1', mu=0, sd=10)

    # this doesn't work:
    alpha = pm.Uniform('alpha', -10, 10)
    beta0 = pm.Uniform('beta0', -10, 10)
    beta1 = pm.Uniform('beta1', -10, 10)
    
    # Expected value of outcome
    mu = alpha + beta0 * X1 + beta1 * X2

    # Likelihood (sampling distribution) of observations
    Y_obs = pm.Normal('Y_obs', mu=mu, sd=1, observed=Y)

start = {'alpha': 0.7, 'beta0': 1.2, 'beta1': 2.1}
map_estimate = pm.find_MAP(model=basic_model, start=start)
map_estimate
```

Traceback (most recent call last):
  File "bug_find_map.py", line 45, in <module>
    map_estimate = pm.find_MAP(model=basic_model, start=start)
  File "/home/david/soft/anaconda3/lib/python3.6/site-packages/pymc3/tuning/starting.py", line 101, in find_MAP
    r = fmin(logp_o, bij.map(start), fprime=grad_logp_o, callback=callback, *args, **kwargs)
  File "/home/david/soft/anaconda3/lib/python3.6/site-packages/pymc3/blocking.py", line 63, in map
    apt[slc] = dpt[var].ravel()
KeyError: 'beta1_interval__'

### PR DESCRIPTION
**Pull request by junpenglao**
find_MAP bug if model contains transform RV but the start value is provided as non-transform.
Using the soft update function from pymc3.sampling

### CODE REVIEW

**Comment by junpenglao** [comment]
Should I change the _update_start_vals into a more general function since we are using it more than one place now?
@fonnesbeck @ColCarroll @twiecki

**Comment by junpenglao** [comment]
This is ready to merge.

File: pymc3/util.py [hunk 1]
```diff
@@ -93,3 +93,18 @@ def get_variable_name(variable):
             return asscalar(value)
         return 'array'
     return name
+
+def update_start_vals(a, b, model):
+    """Update a with b, without overwriting existing keys. Values specified for
+        transformed variables on the original scale are also transformed and inserted.
+        """
```
Comment (twiecki) [inline pymc3/util.py]: wrong indent

File: pymc3/util.py [hunk 2]
```diff
@@ -93,3 +93,18 @@ def get_variable_name(variable):
             return asscalar(value)
         return 'array'
     return name
+
+def update_start_vals(a, b, model):
+    """Update a with b, without overwriting existing keys. Values specified for
```
Comment (twiecki) [inline pymc3/util.py]: blank line after """

File: pymc3/util.py [hunk 3]
```diff
@@ -94,17 +97,20 @@ def get_variable_name(variable):
...
-                    transform_func = [d.transformation for d in model.deterministics if d.name == name]
+                    transform_func = [
+                        d.transformation for d in model.deterministics if d.name == name]
                     if transform_func:
-                        b[tname] = transform_func[0].forward_val(a[name], point=b).eval()
+                        b[tname] = transform_func[0].forward_val(
+                            a[name], point=b).eval()
 
-    a.update({k: v for k, v in b.items() if k not in a})
+    a.update({k: v for k, v in b.items() if k not in a})
```
Comment (twiecki) [inline pymc3/util.py]: missing new line

**Comment by twiecki** [comment]
Thanks @junpenglao.

**Comment by junpenglao** [comment]
Thanks for reviewing Tom ;-)

**Pull request merged**